selfdrive/car/body/carcontroller.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+from selfdrive.car.body import bodycan
+from opendbc.can.packer import CANPacker
+MAX_TORQUE = 500
+MAX_TORQUE_RATE = 50
+MAX_ANGLE_ERROR = 7
+
+class CarController():
+  def __init__(self, dbc_name, CP, VM):
+    self.CP = CP
+    self.car_fingerprint = CP.carFingerprint
+
+    self.lkas_max_torque = 0
+    self.last_angle = 0
+
+    self.packer = CANPacker(dbc_name)
+    # ////////////////////////////////
+    self.i_speed = 0
+
+    self.i_balance = 0
+    self.d_balance = 0
+
+    self.i_torque = 0
+
+    self.speed_measured = 0.
+    self.speed_desired = 0.
+
+    self.torque_r_filtered = 0.
+    self.torque_l_filtered = 0.
+    # ////////////////////////////////
+
+  @staticmethod
+  def deadband_filter(torque, deadband):
+      if torque > 0:
+        torque += deadband
+      else:
+        torque -= deadband
+      return torque
+
+  def update(self, CC, CS, frame, actuators, cruise_cancel, hud_alert,
```

Comment:
Try to use the new CarController conventions, like here: https://github.com/commaai/openpilot/commit/7a2f3c023e0054d1a8fd64b004f8185611323de6